### ISSUE TITLE
bug: navigating with multiple split panes causes first menu to never be reactivated

### ISSUE THREAD

**Issue by yogeshshsf** [ionic-team/ionic-framework#18683]
Hi. i am using 2 split-pane, firstmenu/firstpage will be my home screen. from firstpage i am navigating to secondmenu/secondpage. now when i return back to firstmenu/firstpage, the split-pane is getting hide, can someone help me on this please. see the screenshot and code below.

firstmenu/firstpage: 

![image](https://user-images.githubusercontent.com/48043733/60429106-f7f94480-9c17-11e9-8fb8-ccf2e3c26413.png)

secondmenu/secondpage:

![image](https://user-images.githubusercontent.com/48043733/60429305-6fc76f00-9c18-11e9-9a49-4f23971ad7ed.png)

firstmenu/firstpage:

![image](https://user-images.githubusercontent.com/48043733/60429364-8b327a00-9c18-11e9-9774-b98073a9718d.png)

Below is the github repository to test this:

[https://github.com/yogeshshsf/SplitpaneDemo](url)

**Comment by yogeshshsf** [comment]
Hi. i am using 2 split-pane, firstmenu/firstpage will be my home screen. from firstpage i am navigating to secondmenu/secondpage. now when i return back to firstmenu/firstpage, the split-pane is getting hide, can someone help me on this please. see the screenshot and code below.

Members
 0
18 posts
Report post 
Posted May 29 (edited)
Hi simon. i am using 2 split-pane, firstmenu/firstpage will be my home screen. from firstpage i am navigating to secondmenu/secondpage. now when i return back to firstmenu/firstpage, the split-pane is getting hide. this is stopper for my application, can you help on this please. see the screenshot and code below.﻿﻿

firstmenu/firstpage: 

![image](https://user-images.githubusercontent.com/48043733/60429106-f7f94480-9c17-11e9-8fb8-ccf2e3c26413.png)

**Comment by liamdebeasi** [comment]
Thanks for the issue. I think this problem stems from the fact that we only allow 1 menu to be enabled at a time. If you want these two side panes to be always be open, I would recommend wrapping them in something other than `ion-menu`.

In your demo repo, I wrapped them each in a `div` rather than an `ion-menu` and the issue disappeared. Can you try this and let me know if it works for your use case?

If you really need to use `ion-menu`, I would recommend only having 1 instance of the menu.

**Comment by ErlendEllingsen** [comment]
@liamdebeasi This behaviour also occurs in my app. The split pane shows correctly with two side menus on initial load, but disappears once navigation has been done. On mobile the expected behaviour is of course only to display one menu at the time, but at large desktop screens they both should display as before(?).

**Comment by eduardoRoth** [comment]
Issue happens even if only using one `ion-menu`; @liamdebeasi your quick fix works, if I replace `ion-menu` tag with a `div` one, split pane is always shown.

**Comment by yogeshshsf** [comment]
@liamdebeasi thanks for you help, its much appreciated. it works perfect when you use div..

**Comment by eduardoRoth** [comment]
When using `div` tag instead of `ion-menu`; tags `ion-content` and `ion-footer` don't work as expected.

**Comment by iamkinetic** [comment]
@liamdebeasi are you saying we should have no more than one menu in an application or we may run into problems?

**Comment by liamdebeasi** [comment]
No, but in the case of the original issue the way the multiple menus were added seemed to cause issues. We are investigating this bug and will post an update here when we have an update.

**Comment by daveshirman** [comment]
@yogeshshsf Any luck on this?

**Comment by yogeshshsf** [comment]
@daveshirman  it is working as expected when i wrap using <div> as mentioned by @liamdebeasi comment

**Comment by Fab4guy** [comment]
I have exactly the same problem.

I want to outsource my navigation in a layout component.
I have done all that so far. For testing I created two pages, Home and Contact.
When I load the application, everything is displayed correctly. If I now click on the navigation point Contact, the page changes as expected. If I click on the Home menu item again, the page changes, but the navigation on the left side disappears forever.

I hope you can help me

Here the structure of the different files. I also attach a picture of the folder structure

![folder](https://user-images.githubusercontent.com/18136973/86771531-aec1a900-c052-11ea-9666-4c1b7140a81c.jpg)

**Sorry but somehow the editor does not want to highlight my code**

**app-routing.module.ts**

`import { NgModule } from '@angular/core';
import { PreloadAllModules, RouterModule, Routes } from '@angular/router';
import {MainLayoutWithSideNavComponent} from './layout/main-layout-with-side-nav/main-layout-with-side-nav.component';

const routes: Routes = [
  {
    path: '',
    redirectTo: 'home',
    pathMatch: 'full',
    canActivate: [],
  },
  {
    path: 'home',
    component: MainLayoutWithSideNavComponent,
    canActivate: [],
    loadChildren: () => import('./pages/home/home.module').then(m => m.HomePageModule)
  },
  {
    path: 'contact',
    component: MainLayoutWithSideNavComponent,
    canActivate: [],
    loadChildren: () => import('./pages/contact/contact.module').then(m => m.ContactPageModule)
  }
];

@NgModule({
  imports: [
    RouterModule.forRoot(routes, { preloadingStrategy: PreloadAllModules })
  ],
  exports: [RouterModule]
})
export class AppRoutingModule {}
`
**main-layout-with-side-nav.component.html**

`<ion-split-pane contentId="main-content">

  <ion-menu content-id="main-content">
    <ion-header>
      <ion-toolbar color="primary">
        <ion-title>Menu</ion-title>
      </ion-toolbar>
    </ion-header>

    <ion-content>
      <ion-list>
        <ion-list-header>
          Navigate
        </ion-list-header>
        <ion-menu-toggle auto-hide="false">

          <ion-item *ngFor="let nav of navi" button [routerLink]="'/'+ nav.link">
            <ion-icon slot="start"></ion-icon>
            <ion-label>
              {{nav.linkName}}
            </ion-label>
          </ion-item>

        </ion-menu-toggle>
      </ion-list>
    </ion-content>
  </ion-menu>

  <div class="ion-page" id="main-content">
    <ion-header>
      <ion-toolbar>
        <ion-buttons slot="start">
          <ion-menu-toggle>
            <ion-button>
              <ion-icon slot="icon-only" name="menu"></ion-icon>
            </ion-button>
          </ion-menu-toggle>
        </ion-buttons>
        
      </ion-toolbar>
    </ion-header>
    <ion-content class="ion-padding">
      <ion-router-outlet></ion-router-outlet>
    </ion-content>
  </div>

</ion-split-pane>`

**app.component.html**

`<ion-app>
  <ion-router-outlet animated="false"></ion-router-outlet>
</ion-app>`


@liamdebeasi liamdebeasi is there a solution?

**Comment by thomaux** [comment]
I fixed this by replacing the `<ion-menu>` with a `div` as mentioned before. Then, in order to make `<ion-header>` and `<ion-content>` work properly inside that div, I applied `flex-direction: column;` to the div. Hope this helps!

**Comment by cra1gs** [comment]
> I fixed this by replacing the `<ion-menu>` with a `div` as mentioned before. Then, in order to make `<ion-header>` and `<ion-content>` work properly inside that div, I applied `flex-direction: column;` to the div. Hope this helps!

If you use this method, how does your menu toggle work on smaller screens? My burger menu will not appear, so you cannot access the menu.

**Comment by thomaux** [comment]
@cra1gs this is how I ended up using the split-pane:

```html
<ion-split-pane when="xs" contentId="location-specifics" [disabled]="closeMenu$ | async">
  <div>
    <ion-content>
      <ion-list>
        ...menu content here
      </ion-list>
    </ion-content>
  </div>

  <ion-router-outlet id="location-specifics"></ion-router-outlet>
</ion-split-pane>
```

```typescript
export class SplitPanePage {

  closeMenu$ = this.store.selectedItem$.pipe(
    map((val) => val && window.innerWidth <= 576),
  );
}
```

```sass
@media (max-width: 576px) {
  ion-split-pane {
    --side-min-width: 100%;
  }
}
```

The basic premise of the above is: when an item is selected AND the screen is smaller than the SM breakpoint (=576px), disable the split screen so the "menu" is not shown. In case no item is selected, only the menu is shown as the `when` property is set to `xs` and I've set the side minimum width to 100% for screen widths lower then the SM breakpoint

**Comment by liamdebeasi** [comment]
Hi everyone,

Here is a dev build with a proposed fix if anyone is interested in testing: `7.5.1-dev.11697568647.1ac87d08`

Install Example: `npm install @ionic/angular@7.5.1-dev.11697568647.1ac87d08`
